fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Body.kt

```diff
@@ -0,0 +1,133 @@
+package com.github.kittinunf.fuel.core
+
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.InputStream
+import java.io.OutputStream
+import java.nio.charset.Charset
+
+typealias BodySource = (() -> InputStream)
+typealias BodyLength = (() -> Number)
```

Comment:
[Nit] Can we just use Long?

Reply:
Yes, although it forces callers to cast when a simple int would do.